packages/dart/npt_flutter/lib/features/profile/view/pin_favorites_switch.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:npt_flutter/features/profile_list/bloc/profile_list_bloc.dart';
+import 'package:npt_flutter/features/settings/bloc/settings_bloc.dart';
+import 'package:npt_flutter/styles/sizes.dart';
+
+class PinFavoritesSwitch extends StatelessWidget {
+  const PinFavoritesSwitch({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocListener<SettingsBloc, SettingsState>(
+      // ONLY listen when pinFavorites changes
+      listenWhen: (previous, current) {
+        if (previous is SettingsLoaded && current is SettingsLoaded) {
+          return previous.settings.pinFavorites !=
+              current.settings.pinFavorites;
+        }
+        return false;
```

Comment:
Why a backwards guard-clause instead of this:
```dart
listenWhen: (previous, current) {
  return previous is SettingsLoaded && current is SettingsLoaded &&
    previous.settings.pinFavorites != current.settings.pinFavorites;

},
```

Reply:
I made the change.